src/core/openthread.cpp

```diff
@@ -1256,9 +1256,17 @@ void otSetReceiveIp6DatagramFilterEnabled(otInstance *aInstance, bool aEnabled)
 ThreadError otSendIp6Datagram(otInstance *aInstance, otMessage aMessage)
 {
     ThreadError error;
+    Message *message;
+    Ip6::Header header;
 
     otLogFuncEntry();
 
+    message = static_cast<Message *>(aMessage);
+    assert(message->GetLength() >= sizeof(header));
+
+    message->Read(0, sizeof(header), &header);
+    assert(!header.GetDestination().IsRealmLocalMulticast() || header.GetNextHeader() != Ip6::kProtoHopOpts);
```

Comment:
I see two solutions and i vote just for second since 1st looks for me like a overhead:

1. Scan for any potential MPL Option and as you said replace it with PadN and Pad1 options.
2. Make optimistic assumption that Host will never send MPL Option (why should it?).

The only case that bother me is communication between two separate Thread Networks. Let's say that node from Network A wants to send packet to all node in Network B (let's say the address is FF05::1). Is it possible that MPL option will be propagated between two Backbone routers or only encapsulated IPv6 Header with FF05::1 destination address will be transmitted and encapsulating IPv6 header with MPL option will be just dropped?